src/graph/graph.c

```diff
@@ -556,13 +556,9 @@ void Graph_GetNodeEdges(const Graph *g, const Node *n, GRAPH_EDGE_DIR dir, int e
             GxB_MatrixTupleIter_new(&tupleIter, M);
             GxB_MatrixTupleIter_iterate_row(tupleIter, destNodeID);
         } else {
-            // TODO: Callers wishing to get Incoming edges to a number of nodes
-            // should pass a transposed matrix, as the operations below are costly
-            // and we'll perform them forevery node.
             size_t nRows = Graph_RequiredMatrixDim(g);
             GrB_Vector_new(&incoming, GrB_BOOL, nRows);
             GrB_Descriptor_new(&desc);
-            GrB_Descriptor_set(desc, GrB_INP0, GrB_TRAN);
             GrB_Col_extract(incoming, NULL, NULL, M, GrB_ALL, nRows, destNodeID, desc);
```

Comment:
This is an expensive operation, it requires a full scan over the matrix.
I think it would be much cheaper and faster to extract row destNodeID from the transposed adjacency matrix using  GrB_Col_extract with a descriptor performing transpose, think of it as treating GrB_Col_extract as GrB_Row_extract if the matrix is in CSR format GraphBLAS shouldn't perform the transpose.
Now that we have a vector representing each node with an outgoing edge into `n` we still need to perform some filtering, including only edges of the specified relationship type `R`, and so for each entry in the vector, see if `R[j, n]` is "on" if so keep `j` otherwise drop it.

Reply:
Ahh, that's a good idea! (I need to start working to get into the headspace that column operations are expensive and row operations are cheap haha).

Your suggestion sounds more efficient than the transpose descriptor we had here prior. Are there other areas that could benefit from this logic that would justify extending the interface, or should it be scoped to this function?

Reply:
I'm not sure, I'll have to review graph.c, for now let's start with this function.